### PR TITLE
[SYCL][ESIMD] Add tests for accessor-based gather/scatter and scalar I/O

### PR DESCRIPTION
**Pull request by kbobrovs**
Implementation: https://github.com/intel/llvm/pull/2700

Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: SYCL/ESIMD/accessor_load_store.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
+//==------- accessor_gather_scatter.cpp  - DPC++ ESIMD on-device test ------==//
```
Comment (kychendev) [inline SYCL/ESIMD/accessor_load_store.cpp]: Update file name.

File: SYCL/ESIMD/accessor_load_store.cpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: %ESIMD_RUN_PLACEHOLDER %t.out
```
Comment (kychendev) [inline SYCL/ESIMD/accessor_load_store.cpp:11]: TODO to enable host test? Is there an issue filed for adding emulation support?

Comment (kbobrovs) [reply SYCL/ESIMD/accessor_load_store.cpp:11]: We don't plan to support ESIMD on host device, only CPU emulation device (which appears as GPU device to SYCL RT). 

> Is there an issue filed for adding emulation support?

Do you mean for this particular test? Or for the feature in general?

File: SYCL/ESIMD/accessor_load_store.cpp (hunk 1 above)
Comment (kbobrovs) [reply SYCL/ESIMD/accessor_load_store.cpp]: ok

File: SYCL/ESIMD/accessor_load_store.cpp (hunk 2 above)
Comment (kychendev) [reply SYCL/ESIMD/accessor_load_store.cpp:11]: I mean emulation support for the feature in general.

Comment (kbobrovs) [reply SYCL/ESIMD/accessor_load_store.cpp:11]: Not sure why this issue is needed and how this is related to this PR?

Comment (kychendev) [reply SYCL/ESIMD/accessor_load_store.cpp:11]: Not required for this PR. Just for reference.

**Review by kychendev — APPROVED** [review]

**Comment by vladimirlaz** [comment]
Doing compatibility testing. Please do not merge.

**Pull request merged**